### PR TITLE
added opensea-solana-transfers (Dune v2)

### PR DESCRIPTION
**Pull request by sohwak**
Brief comments on the purpose of your changes:

re-submit PR https://github.com/duneanalytics/abstractions/pull/1358 for Dune v2 only:

*For Dune Engine V2*
I've checked that:

* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`

When you are ready for a review, tag duneanalytics/data-experience. We will re-open your forked pull request as an internal pull request. Then your spells will run in dbt and the logs will be avaiable in Github Actions DBT Slim CI. This job will only run the models and tests changed by your PR compared to the production project.

### CODE REVIEW

**Comment by jeff-dude** [comment]
continuing the convo on this PR, i think we can explore a separate table for this build considering it's solana data and large volumes & the web app timing out on any decent sample size of data. over time, i'd like to avoid getting into this habit, especially as we continue to build upon and improve the performance of queries. we should be able to run subsets of data manually in tests prior to submission and feel comfortable with results.

i'll continue with review of this PR

File: spellbook/models/opensea/solana/opensea_solana_schema.yml [hunk 1]
```diff
@@ -67,4 +67,94 @@ models:
...
+    meta:
+      blockchain: solana
+      project: opensea
+      contributors: sohwak
+    config:
+      tags: ['opensea','solana','events']
+    description: >
+        OpenSea transfers on Solana
+    columns:
+      - name: blockchain
```
Comment (jeff-dude) [inline spellbook/models/opensea/solana/opensea_solana_schema.yml:83]: for any column which has been declared previously in the same file, you can simplify it like lines 60-70 above:
for example: - *blockchain on this column

then you can apply to others below which are reused

File: spellbook/models/opensea/solana/opensea_solana_schema.yml [hunk 2]
```diff
@@ -67,4 +67,94 @@ models:
...
+      - *unique_trade_id
+
+
+  - name: opensea_solana_transfers
+    meta:
+      blockchain: solana
+      project: opensea
+      contributors: sohwak
+    config:
+      tags: ['opensea','solana','events']
```
Comment (jeff-dude) [inline spellbook/models/opensea/solana/opensea_solana_schema.yml:79]: update this tag to 'transfers'

**Review by jeff-dude — CHANGES_REQUESTED** [review]

File: spellbook/models/opensea/solana/opensea_solana_transfers.sql [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+                else 'Other'
+           end purchase_method
+          ,case when array_contains(log_messages, 'Program log: Instruction: Sell') then instructions[6]
+                when array_contains(log_messages, 'Program log: Instruction: Buy') then instructions[7]
+                else instructions[7]
+           end inst
+      from {{ source('solana','transactions') }}
+      where 1=1
+      and success
+      {% if is_incremental() %}
```
Comment (jeff-dude) [inline spellbook/models/opensea/solana/opensea_solana_transfers.sql:25]: fyi, i updated the incremental logic. using this approach vs. now() allows more flexibility if there are errors and data falls behind more than two days

File: spellbook/models/opensea/solana/opensea_solana_transfers.sql [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+           ,royalty_amount_raw / 1e9 * p.price as royalty_fee_amount_usd
+           ,royalty_amount_raw / original_amount_raw * 100 as royalty_fee_percentage
+           ,royalty_receiver_cnt
+           ,royalty_list
+           ,id as unique_trade_id
+      from iv_raw2 a
+      left join {{ source('prices', 'usd') }} p on p.minute = date_trunc('minute', block_time)
+            and p.symbol = 'SOL'
+            {% if is_incremental() %}
+            AND p.minute > (select max(block_time) from {{ this }}) - interval 2 days
```
Comment (jeff-dude) [inline spellbook/models/opensea/solana/opensea_solana_transfers.sql:131]: also added the filter to prices table

**Comment by jeff-dude** [comment]
other general comments:
- i compiled the code and ran query on dune, filtered to one day, seems to run fine
- since this PR is intended to build a short-term object in the database, mainly for deeper analysis on your side, we can skip the steps for adding to files in macros directory (alter table properties, optimize tables) and also tests/seed files. i do expect in the next PR when finalizing opensea solana data that we take these steps to fully build out the pipeline properly

with that said, after the minor fixes to my other comments, i think we may be able to go forward with deploying this

**Comment by sohwak** [comment]
Sorry, I'm late I was on summer break for days. I'm following up now and revising the requested part.

File: spellbook/models/opensea/solana/opensea_solana_schema.yml (hunk 1 above)
Comment (sohwak) [reply spellbook/models/opensea/solana/opensea_solana_schema.yml]: I changed the referenced column like `-*blockchain`, but didn't change the column order so it seems to be mixed.

File: spellbook/models/opensea/solana/opensea_solana_transfers.sql (hunk 4 above)
Comment (sohwak) [reply spellbook/models/opensea/solana/opensea_solana_transfers.sql:131]: Thank you. confirmed.

File: spellbook/models/opensea/solana/opensea_solana_schema.yml (hunk 2 above)
Comment (sohwak) [reply spellbook/models/opensea/solana/opensea_solana_schema.yml]: Thank you. updated.

**Comment by sohwak** [comment]
I think I revised all the parts you mentioned. Please let me if there is any other missed one. Thank you.

**Comment by jeff-dude** [comment]
> d all the parts you mentioned. Please let me if there is any other miss

thank you for revisiting. i'll review this early next week

**Comment by jeff-dude** [comment]
this model can be queried and tested here:
```
select count(1)
  , min(block_date) as first_transfer
  , max(block_date) as most_recent_transfer
from dbt_jeff_opensea_solana.transfers
;
```

**Comment by jeff-dude** [comment]
@sohwak i know you're actively testing this test table here, but wanted to follow up with similar comments from masquot on your other PRs about the repo structure changes. when this code is ready to be submitted for real, please be sure to have your fork and local env up-to-date and ready to merge 👍

**Pull request closed without merging**